src/plugins/lua/luaplugin.h

```diff
@@ -79,9 +70,66 @@ class LUASHARED_EXPORT LuaPlugin : public Tiled::WritableMapFormat
     void writePolygon(LuaTableWriter &, const Tiled::MapObject *);
     void writeTextProperties(LuaTableWriter &, const Tiled::MapObject *);
 
-    QString mError;
-    QDir mMapDir;     // The directory in which the map is being saved
+protected:
+    QDir mMapDir;
     Tiled::GidMapper mGidMapper;
+
 };
 
+class LUASHARED_EXPORT LuaPlugin : public Tiled::Plugin
+{
+    Q_OBJECT
+    Q_INTERFACES(Tiled::Plugin)
+    Q_PLUGIN_METADATA(IID "org.mapeditor.Plugin" FILE "plugin.json")
+
+public:
+    void initialize() override;
+};
+
+
+class LUASHARED_EXPORT LuaMapFormat : public Tiled::WritableMapFormat, public LuaUtilWriter
```

Comment:
Please instantiate the `LuaUtilWriter` on the stack instead of inheriting from it. At least currently I don't see the reason for inheritance. Then you also avoid naming conflicts to you can drop the "Lua" from "writeMapLua" and "writeTilesetLua".

Reply:
Done

Inheritance was just a quick way of sharing the implementation across the two classes. It felt natural given that the methods were all pulled directly from the pevious objects private implementation. Composition works just as well though so I've done as requested.